packages/@n8n/task-runner/src/error-reporter.ts

```diff
@@ -0,0 +1,91 @@
+import { RewriteFrames } from '@sentry/integrations';
+import { init, setTag, captureException, close } from '@sentry/node';
+import type { ErrorEvent, EventHint } from '@sentry/types';
+import * as a from 'assert/strict';
+import { createHash } from 'crypto';
+import { ApplicationError } from 'n8n-workflow';
+
+import type { SentryConfig } from '@/config/sentry-config';
+
+/**
+ * Handles error reporting using Sentry
+ */
+export class ErrorReporter {
+	private isInitialized = false;
+
+	/** Hashes of error stack traces, to deduplicate error reports. */
+	private readonly seenErrors = new Set<string>();
+
+	private get dsn() {
+		return this.sentryConfig.sentryDsn;
+	}
+
+	constructor(private readonly sentryConfig: SentryConfig) {
+		a.ok(this.dsn, 'Sentry DSN is required to initialize Sentry');
+	}
+
+	async start() {
+		if (this.isInitialized) return;
+
+		// Collect longer stacktraces
+		Error.stackTraceLimit = 50;
+
+		process.on('uncaughtException', (error) => {
+			captureException(error);
+		});
+
+		const enabledIntegrations = [
+			'InboundFilters',
+			'FunctionToString',
+			'LinkedErrors',
+			'OnUnhandledRejection',
+			'ContextLines',
+		];
+
+		setTag('server_type', 'task_runner');
+
+		init({
+			dsn: this.dsn,
+			release: this.sentryConfig.n8nVersion,
+			environment: this.sentryConfig.environment,
+			enableTracing: false,
+			serverName: this.sentryConfig.deploymentName,
+			beforeBreadcrumb: () => null,
+			beforeSend: (event, hint) => this.beforeSend(event, hint),
+			integrations: (integrations) => [
+				...integrations.filter(({ name }) => enabledIntegrations.includes(name)),
+				new RewriteFrames({ root: process.cwd() }),
+			],
+		});
+
+		this.isInitialized = true;
+	}
+
+	async stop() {
+		if (!this.isInitialized) {
+			return;
+		}
+
+		await close(1000);
+	}
+
+	beforeSend(event: ErrorEvent, { originalException }: EventHint) {
+		if (!originalException) return null;
+
+		if (originalException instanceof ApplicationError) {
+			const { level, extra, tags } = originalException;
+			if (level === 'warning') return null;
+			event.level = level;
+			if (extra) event.extra = { ...event.extra, ...extra };
+			if (tags) event.tags = { ...event.tags, ...tags };
+		}
+
+		if (originalException instanceof Error && originalException.stack) {
+			const eventHash = createHash('sha1').update(originalException.stack).digest('base64');
+			if (this.seenErrors.has(eventHash)) return null;
+			this.seenErrors.add(eventHash);
+		}
+
+		return event;
+	}
```

Comment:
We're missing the promise check:

https://github.com/n8n-io/n8n/blob/master/packages/cli/src/error-reporting.ts#L72-L74

Reply:
Good catch!